WordPress/gutenberg
# WIP Add/text orientation block support

Depends on:

- https://github.com/WordPress/gutenberg/pull/38883

## Description

This is a WIP/experiment to add a new `writingMode` block support.

<img width="818" alt="Screen Shot 2022-01-28 at 6 26 13 pm" src="https://user-images.githubusercontent.com/6458278/151505565-795042a2-5f06-4aba-a242-2e7a13022fbc.png">

Reference: https://developer.mozilla.org/en-US/docs/Web/CSS/writing-mode

Initially we're adding it to permit design goals, with the layout/i18n aspects to come.

See also:
- https://github.com/WordPress/gutenberg/pull/38816


![Jan-28-2022 18-27-56](https://user-images.githubusercontent.com/6458278/151505550-accd221d-cc33-45a8-9ad4-cecae84e7168.gif)

## TODO

- [x] Depends on the compat work in https://github.com/WordPress/gutenberg/pull/38625
- [ ] Consider creating custom slotfills for the paragraph/heading blocks instead of a fully-fledged member of typography supports (?)


## Testing Instructions

Opt-in to `writingMode` support in `theme.json`

```json
"settings": {
    ....
    "typography": {
       ...
        "writingMode": true,
```

In the site and post editors, insert a paragraph block, then toggle the "Orientation" controls in the Typography panel.

<img width="531" alt="Screen Shot 2022-02-18 at 2 15 24 pm" src="https://user-images.githubusercontent.com/6458278/154610776-6a5b544d-600f-4e66-a685-006fd37aeadc.png">

Ensure what you see in the editors is what you get in the frontend.



## Screenshots 

## Types of changes





## Checklist:
